newIDE/app/src/AssetStore/PrivateAssets/PrivateAssetPackPurchaseDialog.js

```diff
@@ -0,0 +1,291 @@
+          }
+        }
+      }
+    },
+    [
+      loadedReceivedAssetPackInStore,
+      privateAssetPackListingData,
+      isBuying,
+      onClose,
+      isCheckingPurchasesAfterLogin,
+      purchaseSuccessful,
+    ]
+  );
+
+  const dialogContents = !profile
+    ? {
+        subtitle: <Trans>Log-in to purchase this item</Trans>,
+        content: (
+          <CreateProfile
+            onLogin={onLogin}
+            onCreateAccount={onCreateAccount}
+            message={
+              <Trans>
+                Asset packs will be linked to your user account and available
+                for all your Projects. Log-in or Sign-up to purchase this pack.
+              </Trans>
+            }
+            justifyContent="center"
+          />
+        ),
+      }
+    : purchaseSuccessful
+    ? {
+        subtitle: <Trans>Your purchase has been processed!</Trans>,
+        content: (
+          <Line justifyContent="center" alignItems="center">
+            <Text>
+              <Trans>
+                You can close this window and go back to the asset store to
+                download your assets.
+              </Trans>
+            </Text>
+          </Line>
+        ),
+      }
+    : isBuying
+    ? {
+        subtitle: <Trans>Complete your payment on the web browser</Trans>,
+        content: (
+          <>
+            <Line justifyContent="center" alignItems="center">
+              <CircularProgress size={20} />
+              <Spacer />
+              <Text>Waiting for the purchase confirmation...</Text>
+            </Line>
+            <Spacer />
+            <Line justifyContent="center">
+              <BackgroundText>
+                <Trans>
+                  Once you're done, come back to GDevelop and the assets will be
+                  added to your account automatically.
+                </Trans>
+              </BackgroundText>
+            </Line>
+          </>
+        ),
+      }
+    : isCheckingPurchasesAfterLogin
+    ? {
+        subtitle: <Trans>Loading your profile...</Trans>,
+        content: (
+          <Line justifyContent="center" alignItems="center">
+            <CircularProgress size={20} />
+          </Line>
+        ),
+      }
+    : {
+        subtitle: (
+          <Trans>
+            The Asset Pack {privateAssetPackListingData.name} will be linked to
+            your account {profile.email}
+          </Trans>
+        ),
+        content: (
+          <Line justifyContent="center" alignItems="center">
+            <Text>
+              <Trans>
+                A new secure window will open to complete the purchase.
+              </Trans>
+            </Text>
+          </Line>
+        ),
+      };
+
+  const showContinueButton =
+    profile &&
+    !isBuying &&
+    !purchaseSuccessful &&
+    !isCheckingPurchasesAfterLogin;
+  const dialogActions = [
+    <FlatButton
+      key="cancel"
+      label={purchaseSuccessful ? <Trans>Accept</Trans> : <Trans>Cancel</Trans>}
```

Comment:
Looking at the message, I'm not sure that `Accept` is the right word to use here